### DISCUSSION TITLE
自定义节点下foreignObject下body节点是必须的吗？

### DISCUSSION THREAD

**Discussion by GanFeng007** [antvis/X6#1282] [Q&A]
现在使用了layer弹窗，因为layer是通过$('body')来进行追加dom的，就会追加到svg下的body里面去，请问有什么方法避免该body标签吗？目前是改了layer的源码【$('body')=>$(document.body)】来解决的！

**Comment by NewByVector** [comment]
body 不是必须的

**Reply by GanFeng007** [reply]
那么怎么修改呢？

**Reply by NewByVector** [reply]
自定义节点的 markup
